Casks/w/wiso-steuer-2024.rb

```diff
@@ -1,17 +1,23 @@
 cask "wiso-steuer-2024" do
   # NOTE: "2024" is not a version number, but an intrinsic part of the product name
-  version "31.08.4050"
-  sha256 "0b5fb4a649bc7212e77ed5b04dc59368eb7990440f1849dacc3d988f156577d5"
+  version "31.09.4100,-RC1"
+  sha256 "3b21d169989dd7a799c318879839a6f2bfb0e0703021f13a5e9cba3c3d32d897"
 
-  url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/#{version}-HF1/SteuerMac2024-#{version}.dmg",
+  url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Files/#{version.csv.first}#{version.csv.second}/SteuerMac2024-#{version.csv.first}.dmg",
       verified: "update.buhl-data.com/Updates/Steuer/"
   name "WISO Steuer 2024"
   desc "Tax declaration for the fiscal year 2023"
   homepage "https://www.buhl.de/download/wiso-steuer-2024/"
 
   livecheck do
     url "https://update.buhl-data.com/Updates/Steuer/2024/Mac/Aktuell/appcast-steuer.xml"
-    strategy :sparkle
+    regex(%r{/Files/(\d+(?:\.\d+)+)([_-]\w+\d+)/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
+    strategy :sparkle do |item, regex|
+      match = item.url&.match(regex)
+      next if match.blank?
+
+      "#{match[1]},#{match[2]}"
```

Comment:
```suggestion
    regex(%r{/v?(\d+(?:\.\d+)+)([^/]+)?/SteuerMac2024[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
    strategy :sparkle do |item, regex|
      match = item.url&.match(regex)
      next if match.blank?

      match[2] ? "#{match[1]},#{match[2]}" : match[1]
```

The unstable suffix (`-RC1`) won't always be present, so we have to make it optional and handle the presence/absence in the regex/`strategy` block. The existing `url` approach in this PR will work with both, though.